applicationset/services/scm_provider/utils.go

```diff
@@ -165,3 +169,10 @@ func getApplicableFilters(filters []*Filter) map[FilterType][]*Filter {
 	}
 	return filterMap
 }
+
+func checkBranchDoNotExist(err error) bool {
```

Comment:
i think you can just do 
return errors.Is(err, BranchDoNotExist)

Reply:
Yeah sure I would have done it before that looks clean and good

Reply:
nit: I think at this point `checkBranchDoNotExist` is an unnecessary function to have as we can simply check with `errors.Is(err, BranchDoNotExist)`

Reply:
I still feel that function around it would be good and I should comment it so that code is more readable 
WDYT?